### PR TITLE
Initial Yelp library structure

### PR DESCRIPTION
**Pull request by rteague**
- Generated the directory structure for composer
- Added Yelp.php and YelpTest.php

### CODE REVIEW

File: src/Yelp.php [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+namespace Neighborhoods\Libraries;
+
+use Exception;
+use GuzzleHttp\Client;
+
+/**
+ * Class Yelp
+ *
+ * @package Neighborhoods\Libraries
```
Comment (MattStypa) [inline src/Yelp.php]: Wrong namespace. `Neighborhoods\YelpFusion`

File: src/Yelp.php [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+ *
+ * @category Library
+ * @package  Yelp
+ * @author   Matt Stypa <matt.stypa@55places.com>
+ * @author   Rashaud Teague <rashaud.teague@55places.com>
+ * @license  MIT https://opensource.org/licenses/MIT
+ * @link     https://github.com/neighborhoods/Yelp-Fusion
+ */
+
+namespace Neighborhoods\Libraries;
```
Comment (MattStypa) [inline src/Yelp.php]: Neighborhoods\YelpFusion

File: src/Yelp.php [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    const REQUEST_HEADERS = [
+        'cache-control' => 'no-cache',
+    ];
+
+    protected $guzzle;
+
+    /**
+     * Yelp constructor.
+     *
+     * @param null $handler
```
Comment (MattStypa) [inline src/Yelp.php]: Needs more information about this

File: src/Yelp.php [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+        $this->guzzle = new Client($options);
+    }
+
+    /**
+     * Get the bearer token
+     *
+     * @param $clientId
+     * @param $clientSecret
+     * @return mixed
```
Comment (MattStypa) [inline src/Yelp.php]: This returns an object

File: src/Yelp.php [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+            )
+        );
+    }
+
+    /**
+     * Gets the search results
+     *
+     * @param $terms
+     * @param $bearerToken
+     * @return mixed
```
Comment (MattStypa) [inline src/Yelp.php]: Object

File: src/Yelp.php [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+        $this->guzzle = new Client($options);
+    }
+
+    /**
+     * Get the bearer token
+     *
+     * @param $clientId
+     * @param $clientSecret
+     * @return mixed
+     */
```
Comment (MattStypa) [inline src/Yelp.php]: Missing @throws

File: src/Yelp.php [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+        );
+    }
+
+    /**
+     * Gets the search results
+     *
+     * @param $terms
+     * @param $bearerToken
+     * @return mixed
+     */
```
Comment (MattStypa) [inline src/Yelp.php]: Missing @throws

File: src/Yelp.php [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+                ]
+            )
+        );
+    }
+
+    /**
+     * Parses the response (decodes the json response)
+     *
+     * @param $response
+     * @return mixed
```
Comment (MattStypa) [inline src/Yelp.php]: Object

File: src/Yelp.php [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+
+        if ($handler) {
+            $options['handler'] = $handler;
+        }
+
+        $this->guzzle = new Client($options);
+    }
+
+    /**
+     * Get the bearer token
```
Comment (MattStypa) [inline src/Yelp.php]: 'Gets the bearer token'

File: tests/YelpTest.php [hunk 10]
```diff
@@ -0,0 +1,204 @@
+<?php
+/**
+ * Test suite for the Yelp library
+ *
+ * PHP version 5.6
+ *
+ * @category Library
+ * @package  YelpTest
```
Comment (MattStypa) [inline tests/YelpTest.php]: Shouldn't this be `Yelp`

File: composer.json [hunk 11]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (MattStypa) [inline composer.json:1]: This file has mixed indentation. Tabs+Spaces

File: phpunit.xml [hunk 12]
```diff
@@ -0,0 +1,12 @@
+<phpunit 	bootstrap="vendor/autoload.php"
```
Comment (MattStypa) [inline phpunit.xml]: Please use space indentation and align the properties better

File: phpunit.xml [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+			colors="true"
+			convertErrorsToExceptions="true"
+			convertNoticesToExceptions="true"
+			convertWarningsToExceptions="true"
+			stopOnFailure="true">
+			<testsuites>
+				<testsuite name="Unit Tests">
+					<directory>tests</directory>
+				</testsuite>
+			</testsuites>
```
Comment (MattStypa) [inline phpunit.xml]: Missing whitelist configuration

**Review by MattStypa — CHANGES_REQUESTED** [review]

File: src/Yelp.php (hunk 7 above)
Comment (rteague) [reply src/Yelp.php]: But this doesn't throw anything.

Comment (MattStypa) [reply src/Yelp.php]: It calls two separate methods that throw so this throws as well.

File: src/Yelp.php [hunk 14]
```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * A library that interfaces with the Yelp Fusion (v3) API
+ *
+ * PHP version 5.6
+ *
+ * @category Library
+ * @package  Yelp
```
Comment (MattStypa) [inline src/Yelp.php]: `Neighborhoods\YelpFusion`

File: src/Yelp.php [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+        $this->guzzle = new Client($options);
+    }
+
+    /**
+     * Gets the bearer token
+     *
+     * @param $clientId
+     * @param $clientSecret
+     * @return object
+     */
```
Comment (MattStypa) [inline src/Yelp.php]: @throws

File: src/Yelp.php [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+        );
+    }
+
+    /**
+     * Gets the search results
+     *
+     * @param $terms
+     * @param $bearerToken
+     * @return object
+     */
```
Comment (MattStypa) [inline src/Yelp.php]: @throws

File: tests/YelpTest.php [hunk 17]
```diff
@@ -0,0 +1,204 @@
+<?php
+/**
+ * Test suite for the Yelp library
+ *
+ * PHP version 5.6
+ *
+ * @category Library
+ * @package  Yelp
```
Comment (MattStypa) [inline tests/YelpTest.php]: Neighborhoods\YelpFusion

File: composer.json [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+    "php": ">=5.6",
+    "guzzlehttp/guzzle": "~6.0"
+  },
+  "require-dev": {
+    "phpunit/phpunit": "5.5.*",
+    "squizlabs/php_codesniffer": "3.*"
+  },
+  "autoload": {
+    "psr-4": {
+      "Neighborhoods\\": "src/"
```
Comment (MattStypa) [inline composer.json]: "Neighborhoods\\\\YelpFusion\\\\": "src/"

**Review by MattStypa — CHANGES_REQUESTED** [review]

File: tests/YelpTest.php [hunk 19]
```diff
@@ -0,0 +1,204 @@
...
+ * @author   Matt Stypa <matt.stypa@55places.com>
+ * @author   Rashaud Teague <rashaud.teague@55places.com>
+ * @license  MIT https://opensource.org/licenses/MIT
+ * @link     https://github.com/neighborhoods/Yelp-Fusion
+ */
+
+namespace Neighborhoods\Tests\YelpFusion;
```
Comment (MattStypa) [inline tests/YelpTest.php]: Neighborhoods\YelpFusion\tests;

**Review by MattStypa — CHANGES_REQUESTED** [review]

File: src/Yelp.php (hunk 7 above)
Comment (jpmarcotte) [reply src/Yelp.php]: This will probably be helpful: https://blog.jetbrains.com/phpstorm/2016/07/docblock-improvements-for-throws-in-phpstorm-2016-2/

**Review by MattStypa — APPROVED** [review]

**Pull request merged**